enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/MobileTokenController.java

```diff
@@ -115,8 +125,24 @@ public ObjectResponse<OperationListResponse> operationList(@Parameter(hidden = t
                 throw new MobileTokenAuthException();
             }
         } catch (PowerAuthClientException e) {
-            logger.error("Unable to call upstream service.", e);
-            throw new MobileTokenAuthException();
+            final String errorCode = e.getPowerAuthError().map(PowerAuthError::getCode).orElse("ERROR_CODE_MISSING");
+            switch (errorCode) {
+                case APPLICATION_NOT_FOUND -> {
+                    logger.info("Application ID: {} not found: {}", auth.getApplicationId(), e.getMessage());
+                    logger.debug("Application ID: {} not found.", auth.getApplicationId(), e);
+                    throw new MobileTokenException(ErrorCode.INVALID_APPLICATION, "No application was found with the provided identifier.");
+                }
+                case INVALID_REQUEST -> {
+                    logger.info("Request validation error: {}", e.getMessage());
+                    logger.debug("Request validation error.", e);
+                    throw new MobileTokenException(ErrorCode.INVALID_REQUEST, "Request validation error: %s".formatted(e.getMessage()));
+                }
+                default -> {
+                    logger.warn("Calling PowerAuth service failed: {}", e.getMessage());
+                    logger.debug("Calling PowerAuth service failed.", e);
+                    throw new InternalServiceException("Unable to call upstream service.");
```

Comment:
🤔 Is it really `InternalServiceException` or rather `RemoteCommunicationException`? Moreover, we provide details for  the `ERR0015` and `ERR0024` only, other error codes cause `InternalServiceException`, correct? But we map `RemoteCommunicationException` to 409 status code.